fix #285432: save offset property for fermata

peterhieuvu opened the pull request:
Originally, the offset property wasn't saved because Pid::OFFSET is styled property which means it isn't saved by Element::writeProperties(). Therefore all we need to do is save it ourselves :)

MarcSabatella commented:
I think this isn't the right fix, it will cause OFFSET to always be written.  Instead, we need to make sure we set the UNSTYLED property flag in the places where we are changing the offset.  I'm not sure why this isn't happening already, though.

peterhieuvu commented:
Oh you're right that was my mistake. Can't this be adjusted for just with a check for if the OFFSET is styled or not? The property flags are set in the right places, but it's just the since the property is a styled property with default style, it will never be UNSTYLED and so will never be written by Element::writeProperties(). I think it's a bit of a unique case since the other articulations don't have a default style for OFFSET. The code that's normally responsible is at the end of Element::writeProperties() where the code writes a number of common properties if they have property flag NOSTYLE. But in this case, again, OFFSET can't have property style NOSTYLE so a check for whether or not OFFSET is styled should function in the same way. 

When the reset button is pressed, OFFSET is set to STYLED again so with the check for isStyled(Pid::OFFSET) before writing the property, the property is only written when the OFFSET is UNSTYLED which I think is the desired behavior. Thank you!

peterhieuvu commented:
Ahh I made a mistake and forgot to add a bit of code for the compatibility. It should be good now I think.

peterhieuvu commented:
I'm sorry for doing so many forced commits, but I finally think I have fixed the issue. Offset should now be saved only when needed. I ended up modifying one of the tests because I think it was misrepresented. In the fermata test, the reference file actually was a score that neglected the offset in the old score which is not something I think should happen. For that reason, I updated the compatibility test for fermatas so that it included the saving of offsets for fermatas.
